sklearn/covariance/outlier_detection.py

```diff
@@ -36,7 +37,12 @@ class OutlierDetectionMixin(object):
     """
     def __init__(self, contamination=0.1):
         self.contamination = contamination
+        self.threshold_ = None
+        # FIXME to be removed in 0.18
         self.threshold = None
+        warnings.warn(("The threshold attribute is renamed to threshold_ from "
```

Comment:
I thought of `__getattr__` and then warning when `threshold` is accessed... but this is cleaner! ;)